mreg_cli/api/models.py

```diff
@@ -2701,7 +2701,7 @@ def get_by_any_means(
                     return hosts[0]
 
                 if len(hosts) > 1:
```

Comment:
We could do this:

```suggestion
                if len(hosts) > 1 and any(hosts[0].id != host.id for host in hosts):
```

But should we consider refactoring the method used to fetch hosts to `Host.get_list_by_query_unique`, and then instead catch `MultipleEntitiesFound` and raise the more granular `MultipleEntitiesFound` we already have? Something like:


```py
            try:
                ptr = False
                ipaddress.ip_address(identifier)

                host = Host.get_by_query_unique({"ipaddresses__ipaddress": identifier})
                if not host:
                    host = Host.get_by_query_unique({"ptr_overrides__ipaddress": identifier})
                    ptr = True

                if host:
                    if ptr and inform_as_ptr:
                        OutputManager().add_line(f"{identifier} is a PTR override for {host.name}")
                    return host
            except MultipleEntitiesFound as e:
                raise MultipleEntitiesFound(
                    f"Multiple hosts found with IP address or PTR {identifier}."
                ) from e
            except ValueError: # invalid IP (should use NetworkOrIP.parse...)
                pass
```

Reply:
I have pushed a commit which implements this. I do think we need more comprehensive tests for a broader range of host/ip/mac/network permutations though. I'll look into it once we have implemented #323.